From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Mon, 25 Mar 2019 16:58:00 +0100
Subject: [PATCH] media: coda: improve DEC_PIC_SUCCESS reporting

The 0x40001 return value is observed at the end of some MPEG-2 streams,
when no more data is available to decode.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 drivers/media/platform/coda/coda-bit.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/coda/coda-bit.c b/drivers/media/platform/coda/coda-bit.c
index 9130aacd11a6..1521d2cf3d5b 100644
--- a/drivers/media/platform/coda/coda-bit.c
+++ b/drivers/media/platform/coda/coda-bit.c
@@ -2297,8 +2297,12 @@ static void coda_finish_decode(struct coda_ctx *ctx)
 	src_fourcc = q_data_src->fourcc;
 
 	val = coda_read(dev, CODA_RET_DEC_PIC_SUCCESS);
-	if (val != 1)
-		pr_err("DEC_PIC_SUCCESS = %d\n", val);
+	if (val == 0x40001 && src_fourcc == V4L2_PIX_FMT_MPEG2) {
+		coda_dbg(1, ctx, "end of stream signal?\n");
+		/* TODO: stop if no frame finished */
+	} else if (val != 1) {
+		pr_err("DEC_PIC_SUCCESS = 0x%x\n", val);
+	}
 
 	success = val & 0x1;
 	if (!success)
